### PR TITLE
Move cached_revision from downward lab to lab

### PR DESCRIPTION
**Pull request by thomaskeller79**
I added the name of the build script and the exclude dirs as member variables to the CachedRevision super class in lab/cached_revision.py that are set in the constructor, while _cleanup() has no functionality in CachedRevision at all and is overwritten in CachedFastDownwardRevision in downward/cached_revision.py.

### CODE REVIEW

**Comment by gfrances** [comment]
Looks like you broke Lab, Thomas !

**Comment by thomaskeller79** [comment]
> Looks like you broke Lab, Thomas !

Only that was my plan all the time mwahahaha!

Any idea why? I fixed the black error, and flake8 doesn't complain on my machine.

**Comment by jendrikseipp** [comment]
The tests run flake8 with several flake8 extensions. To run all tests use "tox". To run only style tests use "tox -e style".

**Comment by gfrances** [comment]
The error msg is there in the logs hidden somewhere...:

    ./lab/cached_revision.py:128:83: B006 Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.

File: lab/cached_revision.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    return m.hexdigest()[:8]
+
+
+class CachedRevision(object):
+    """This class represents checkouts of the used solver.
+
+    It provides methods for caching and compiling given revisions.
+    """
+
+    def __init__(self, repo, local_rev, build_script, build_options, exclude_dirs=[]):
```
Comment (gfrances) [inline lab/cached_revision.py]: You should prob. use here exclude_dirs=None, and then on the first line of the init method just do `exclude_dirs = [] if exclude_dirs=None else exclude dirs`, or similar

**Comment by jendrikseipp** [comment]
You can also use "tox -e fix-style" to fix some of the errors automatically.

@gfrances could you do the first round of code review once the tests pass, please?

File: lab/cached_revision.py (hunk 1 above)
Comment (jendrikseipp) [reply lab/cached_revision.py]: A more idiomatic solution would be `exclude_dirs = exclude_dirs or []`, but I haven't looked at the code.

Comment (thomaskeller79) [reply lab/cached_revision.py]: I just removed the default, it's not necessary

**Comment by thomaskeller79** [comment]
How can I update a pull request in git?

**Comment by jendrikseipp** [comment]
> How can I update a pull request in git?

Pushing the branch to your repo should be enough.

**Comment by jendrikseipp** [comment]
I compared the two main files locally as you suggested and the changes look good to me. I'll merge this now and will add the documentation in the main branch. Thanks for your work!

**Comment by jendrikseipp** [comment]
For the next PR, I recommend making your changes in a branch and allowing me to push to your branch.

**Pull request merged**